osu.Game/Screens/Play/Player.cs

```diff
@@ -585,6 +583,29 @@ private void performUserRequestedSkip()
         /// <param name="time">The destination time to seek to.</param>
         public void Seek(double time) => GameplayClockContainer.Seek(time);
 
+        private ScheduledDelegate frameStablePlaybackResetDelegate;
+
+        /// <summary>
+        /// Seeks to a specific time in gameplay, bypassing frame stability.
+        /// </summary>
+        /// <remarks>
+        /// Intermediate hitobject judgements may not be applied or reverted correctly during this seek.
+        /// </remarks>
+        /// <param name="time">The destination time to seek to.</param>
+        public void NonFrameStableSeek(double time)
```

Comment:
Not entirely thrilled about this method existing, but I'm not sure there is a way around it. I was going to ask if doing something directly to the underlying clocks would render this unnecessary, but I think that the stability logic will be a blocker to such a solution. Was _also_ going to ask if this could live in `SpectatorPlayer`, but the description answered that this is planned to be used in other non-spectator contexts.

I'd at least propose not making the method not-public, though. Protected/internal should work for now, and I don't really feel implementation details such as this should be overly outwardly exposed until there is an actual good reason to do so.

Reply:
Sure. I have a feeling it will be used in a public context with spectator, but that can happen when it's required.